bitcoin/BRSyncManager.c

```diff
@@ -1093,77 +1114,49 @@ BRClientSyncManagerAnnounceGetTransactionsDone (BRClientSyncManager manager,
         // bwmAnnounceTransaction()  (for each one or with all of them).
         manager->clientCallbacks.funcGetTransactions (manager->clientContext,
                                                       BRClientSyncManagerAsSyncManager (manager),
-                                                      addressStrings,
-                                                      addressCount,
+                                                      (const char **) addresses,
+                                                      array_count(addresses),
                                                       begBlockNumber,
                                                       endBlockNumber,
                                                       rid);
     }
 
-    if (addressStrings) {
-        free (addressStrings);
-    }
-
-    if (addressArray) {
-        free (addressArray);
+    if (NULL != addresses) {
+        array_free (addresses);
```

Comment:
Does each 'char *' address need to be freed? BRSyncManagerGetTransactionsCallback is `OwnershipKept` for `addresses`.

Reply:
In this design, no, the individual address is owned by the scanState's knownAddress set. BUT it occurs to me that this callback occurs outside of the lock, so I will change it to avoid a race here.

Reply:
`OwnershipKept` implies BRSyncManagerGetTransactionsCallback doesn't need to do anything with `addresses`, which is the current design. What am I missing from your comment in that respect?

Reply:
I thought if `BRSyncManagerGetTransactionsCallback()` did nothing (by design) when BRSyncManager would need to do something with the array contents.  You confirmed that BRSyncManager owns them.